lib/pages/logged_in/social_media_page/components/friend_request.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flutter/material.dart';
+import 'package:runningapp/database/repository.dart';
+
+class FriendRequest extends StatefulWidget {
+  final Repository repository;
+  final String name;
+  final String userId;
+  final bool added;
+
+  const FriendRequest({
+    super.key,
+    required this.name,
+    required this.userId,
+    required this.added,
+    required this.repository,
+  });
+
+  @override
+  FriendRequestState createState() => FriendRequestState();
+}
+
+class FriendRequestState extends State<FriendRequest> {
+  bool _isVisible = true;
+
+  @override
+  Widget build(BuildContext context) {
+    if (!_isVisible) {
+      return const SizedBox.shrink();
+    }
+
+    return Card(
+      child: Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: Row(
+          children: [
+            const CircleAvatar(
+              backgroundImage:
+                  NetworkImage('https://picsum.photos/id/237/200/300'),
```

Comment:
Similar to my earlier comment, consider using a placeholder image in the event of failure to retrieve networkImage